arrows/vtk/applets/color_mesh.cxx

```diff
@@ -0,0 +1,527 @@
+    writer->AddInputDataObject(mesh);
+    writer->Write();
+  }
+
+  bool run_algorithm()
+  {
+    // Attempt opening input and output files.
+    //  - filepath validity checked above
+    video_file_ = config_->get_value<std::string>("video_source");
+    mask_file_ = config_->get_value<std::string>("mask_source");
+
+    LOG_INFO(main_logger, "Reading video...");
+    set_video(config_, video_file_);
+    bool hasMask = !mask_file_.empty();
+    if (hasMask)
+    {
+      set_mask(config_, mask_file_);
+    }
+    LOG_INFO(main_logger, "Load camera map...");
+    auto cameras = load_camera_map(video_reader_, video_file_, cameras_dir_);
+    set_cameras(cameras);
+    LOG_INFO(main_logger, "Load mesh file...");
+    auto mesh = load_mesh(input_mesh_file_);
+    set_input(mesh);
+    set_output(mesh);
+    set_frame(frame_);
+    set_frame_sampling(frame_sampling_);
+    set_average_color(average_color_);
+    colorize();
+    LOG_INFO(main_logger, "Save mesh file...");
+    save_mesh(mesh, output_mesh_file_.c_str());
+    return true;
+  }
+
+  void report_progress_changed(
+    std::string const& message, int percentage) override
+  {
+        LOG_INFO(main_logger,
+                 message << ": " << percentage << "%");
+  }
+};
+
+// ----------------------------------------------------------------------------
+int
+color_mesh::
+run()
+{
+
+  try
+  {
+    switch (d->process_command_line(command_args()))
+    {
+    case priv::HELP:
+      std::cout << m_cmd_options->help();
+      return EXIT_SUCCESS;
+    case priv::WRITE:
+      return EXIT_SUCCESS;
+    case priv::FAIL:
+      return EXIT_FAILURE;
+    case priv::SUCCESS:
+      ;
+    }
+
+    if (d->config_ == nullptr)
+    {
+      return EXIT_FAILURE;
+    }
+
+    d->initialize();
+    if (!d->run_algorithm())
+    {
+      return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+  }
+  catch (std::exception const& e)
+  {
+    LOG_ERROR(main_logger, "Exception caught: " << e.what());
+
+    return EXIT_FAILURE;
+  }
+  catch (...)
+  {
+    LOG_ERROR(main_logger, "Unknown exception caught");
+
+    return EXIT_FAILURE;
+  }
+} // run
+
+// ----------------------------------------------------------------------------
+void
+color_mesh::
+add_command_options()
+{
+  m_cmd_options->custom_help( wrap_text(
+    "[options] input-mesh video-file cameras-dir output-mesh\n"
+    "This program colors an input-mesh from a video (or list of images) and "
+    "a list of camera files stored in a directory. A mesh colored with "
+    "the average color or with a color for a particular camera is produced.") );
+
+  m_cmd_options->positional_help("\n  input-mesh  - input mesh file."
+                                 "\n  video-file  - input video file."
+                                 "\n  cameras-dir  - input camera directory."
+                                 "\n  output-mesh - output mesh file."
+  );
+
+  m_cmd_options->add_options()
+    ( "a,average-color",
```

Comment:
Do we have options for median as a config parameter?

Reply:
When that is set, it saves both average and median. We could name it something different?

Reply:
What you're doing seems good to me.

Reply:
The common case is computing the combined (e.g. mean or median) color.  So instead of an option for this I think it should save the combined/average color by default unless you used an option, like `--frame`, that indicates otherwise.  We could replace this option with "a,all-colors" instead and have it save colors of all frames if specified.  This would only work if the output file was a `.vtp`.

Note that when the output is not `.vtp` only one color mode (e.g. median or mean) can be saved.  So we should also have an option or configuration parameter to select which of the color arrays is saved.  In TeleSculptor we save whichever one is selected for visualization.

Reply:
Sounds good.  I used "a,all-frames" and "p,ply_color" where ply_color can be mean or median.